src/extras/amd.js

```diff
@@ -70,104 +62,46 @@ import { errMsg } from '../err-msg.js';
     }
   }
 
-  // hook System.register to know the last declaration binding
-  var lastRegisterDeclare;
-  var systemRegister = systemPrototype.register;
-  systemPrototype.register = function (name, deps, declare) {
-    lastRegisterDeclare = typeof name === 'string' ? declare : deps;
-    systemRegister.apply(this, arguments);
-  };
-
-  var instantiate = systemPrototype.instantiate;
-  systemPrototype.instantiate = function() {
-    // Reset "currently executing script"
-    amdDefineDeps = null;
-    return instantiate.apply(this, arguments);
-  };
-
-  var getRegister = systemPrototype.getRegister;
-  systemPrototype.getRegister = function () {
-    if (tmpRegister)
-      return tmpRegister;
+  global.define = function (arg1, arg2, arg3) {
+    var isNamedRegister = typeof arg1 === 'string';
+    var name = isNamedRegister ? arg1 : null;
+    var depArg = isNamedRegister ? arg2 : arg1;
+    var execArg = isNamedRegister ? arg3 : arg2;
 
-    var _firstNamedDefine = firstNamedDefine;
-    firstNamedDefine = null;
+    // The System.register(deps, exec) arguments
+    var deps, exec;
 
-    var register = getRegister.call(this);
-    // if its an actual System.register leave it
-    if (register && register[1] === lastRegisterDeclare)
-      return register;
-
-    var _amdDefineDeps = amdDefineDeps;
-    amdDefineDeps = null;
-
-    // If the script registered a named module, return that module instead of re-instantiating it.
-    if (_firstNamedDefine)
-      return _firstNamedDefine;
-
-    // otherwise AMD takes priority
-    // no registration -> attempt AMD detection
-    if (!_amdDefineDeps)
-      return register || emptyInstantiation;
-
-    return createAMDRegister(_amdDefineDeps, amdDefineExec);
-  };
-  var amdDefineDeps, amdDefineExec;
-  global.define = function (name, deps, execute) {
-    var depsAndExec;
-    // define('', [], function () {})
-    if (typeof name === 'string') {
-      depsAndExec = getDepsAndExec(deps, execute);
-      if (amdDefineDeps) {
-        if (!System.registerRegistry) {
```

Comment:
This is the only behavior I am aware of that I intentionally changed.

Previously, named defines would sometimes throw an error and sometimes not, depending on whether there were multiple calls to define in the same file. I changed it so that it never throws an error for named defines, thinking that instead we can just warn when that occurs.

Not sure if that would be breaking behavior for anybody.

Reply:
That seems fine.